### DISCUSSION TITLE
Strimzi kafka - Ingress

### DISCUSSION THREAD

**Discussion by k8sapps123** [strimzi/strimzi-kafka-operator#8234] [General]
We are trying to do Ingress for strimzi kafka but everytime kafka pod is not creating where as Zookeeper pod is creating. Below is the configuration used for the same:

Reference doc: https://strimzi.io/blog/2019/05/23/accessing-kafka-part-5/

apiVersion: kafka.strimzi.io/v1beta2
kind: Kafka
metadata:
  name: dev
spec:
  kafka:
    version: 3.2.0
    replicas: 1
    listeners:
      - name: plain
        port: 9092
        type: internal
        tls: false
      - name: tls
        port: 9093
        type: internal
        tls: true
        authentication:
          type: tls
      - name: external
        port: 9094
        type: ingress
        tls: true
        configuration:
          bootstrap:
            host: dev-kafka-bootstrap.dev.sree.com
            annotations:
               external-dns.alpha.kubernetes.io/hostname: dev-kafka-bootstrap.dev.sree.com.
               external-dns.alpha.kubernetes.io/ttl: "60"
          brokers:
          - broker: 0
            host: dev-kafka-broker-0.dev.sree.com
            annotations:
              external-dns.alpha.kubernetes.io/hostname: dev-kafka-broker-0.dev.sree.com.
              external-dns.alpha.kubernetes.io/ttl: "60"
          - broker: 1
            host: dev-kafka-broker-1.dev.sree.com
            annotations:
              external-dns.alpha.kubernetes.io/hostname: dev-kafka-broker-1.dev.sree.com.
              external-dns.alpha.kubernetes.io/ttl: "60"
          - broker: 2
            host: dev-kafka-broker-2.dev.sree.com
            annotations:
               external-dns.alpha.kubernetes.io/hostname: dev-kafka-broker-2.dev.sree.com.
               external-dns.alpha.kubernetes.io/ttl: "60"
    authorization:
      type: simple
    config:
      offsets.topic.replication.factor: 1
      transaction.state.log.replication.factor: 1
      transaction.state.log.min.isr: 1
      default.replication.factor: 1
      min.insync.replicas: 1
      inter.broker.protocol.version: "3.2"
    storage:
      type: ephemeral
  zookeeper:
    replicas: 1
    storage:
      type: ephemeral
  entityOperator:
    topicOperator: {}
    userOperator: {}

Please let us know is there any problem with above yaml? Is there any alternative approach to do ingress for strimzi kafka?

**Comment by scholzj** [comment]
If you format the YAML as a code, it will be readable and it would be possible to say if it looks good. But to be honest, I'm not sure I understood what exactly the issue you have is - can you clarify that? The title suggests you have some Ingress issue. But the first sentence suddenly talks about ZooKeeper.

**Reply by k8sapps123** [reply]
I'm trying to **create ingress for strimzi kafka** by referring to doc:
Reference doc:  https://strimzi.io/blog/2019/05/23/accessing-kafka-part-5/

**With above config file when we apply it, kafka broker pods not creating. Not sure above configuration is correct or not.**
**I doubt that yaml file configuration is wrong. 

Or is there any otherway to do ingress for strimzi kafka?**

**Reply by scholzj** [reply]
It is hard to say if it is wrong or not without proper formatting. But if you are sure it is ok, then please share the logs from the operator and the Ingress resources from your Kubernetes cluster (`kubectl get ingress -o yaml`) as that would be the next step to investigate.
